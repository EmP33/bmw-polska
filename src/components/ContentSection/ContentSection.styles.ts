import styled from "styled-components";

export const ContentSection = styled.section<{ type: "blocks" | "text" }>`
  margin: ${({ type }) => (type === "text" ? "0 auto" : "0 auto 3rem")};
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  width: 92%;
  overflow: hidden;

  @media only screen and (max-width: 1200px) {
    grid-template-columns: 1fr;
  }

  .images {
    display: grid;
    grid-template-rows: repeat(5, 1fr);
    grid-template-columns: repeat(6, 1fr);
    align-items: center;
    column-gap: 1rem;
    grid-row: 1/2;
    grid-column: ${({ type }) => (type === "blocks" ? 1 : 2)};

    @media only screen and (max-width: 1200px) {
      grid-template-rows: ${({ type }) =>
        type === "blocks" ? "repeat(5, 1fr)" : "repeat(4, 1fr)"};
      margin-bottom: ${({ type }) => (type === "blocks" ? "0" : "3rem")};
      grid-row: 1;
      grid-column: 1/-1;
    }

    @media only screen and (max-width: 600px) {
      max-height: 500px;
    }

    img {
      width: 100%;
      height: auto;

      &:nth-child(1) {
        grid-column: ${({ type }) => (type === "text" ? "5/ -1" : "1/3")};
        grid-row: ${({ type }) => (type === "text" ? "1 / 3" : "3/-1")};
      }
      &:nth-child(2) {
        grid-column: ${({ type }) => (type === "blocks" ? "3/ -1" : "1/5")};
        grid-row: ${({ type }) => (type === "blocks" ? "1 / 5" : "2/5")};
      }
    }
  }

  .content {
    padding-left: ${({ type }) => (type === "text" ? "0" : "7.5rem")};
    width: ${({ type }) => (type === "text" ? "536px" : "auto")};
    display: grid;
    align-items: center;
    align-content: center;
    margin: 0 auto;
    grid-column: ${({ type }) => (type === "blocks" ? 2 : 1)};

    @media only screen and (max-width: 1200px) {
      grid-row: 2;
      grid-column: 1/-1;
    }

    @media only screen and (max-width: 900px) {
      padding: 0 0.5rem 0 0.5rem;
    }
    @media only screen and (max-width: 600px) {
      width: auto;
    }

    svg {
      width: ${({ type }) => (type === "text" ? "50px" : "25px")};
      height: ${({ type }) => (type === "text" ? "50px" : "50px")};
      transform: ${({ type }) => (type === "text" ? "rotateY(180deg)" : "0")};
      fill: var(--color-gray-4);
    }
    h2 {
      width: 300px;
      font-size: 2.25rem;
      font-family: var(--font-heading);
      font-weight: bold;
      line-height: 2.5rem;
      margin-bottom: 3rem;

      @media only screen and (max-width: 900px) {
        font-size: 1.5rem;
        width: 200px;
      }
    }

    h4 {
      font-size: 1.25rem;
      font-family: var(--font-primary);
      font-weight: 300;
      color: var(--color-gray-3);
      margin-bottom: 2rem;
    }

    p {
      font-size: 0.875rem;
      font-family: var(--font-primary);
      color: var(--color-gray-1);
      line-height: 1.25rem;
    }

    .content__blocks {
      display: flex;
      flex-wrap: wrap;
      column-gap: 2rem;
      row-gap: 4rem;

      .block {
        height: 120px;
        width: 191px;

        @media only screen and (max-width: 600px) {
          width: 100%;
          height: auto;
        }
        h6 {
          font-size: 0.75rem;
          font-family: var(--font-heading);
          font-weight: bold;
          position: relative;
          padding-bottom: 0.3rem;

          &::before {
            content: "";
            position: absolute;
            bottom: 0;
            left: 0;
            width: 10%;
            height: 2px;
            background: var(--color-gray-2);
            z-index: 2;
          }
          &::after {
            content: "";
            position: absolute;
            bottom: 0;
            left: 0;
            width: 100%;
            height: 2px;
            background: var(--color-gray-6);
          }
        }
        p {
          font-size: 0.875rem;
          color: var(--color-gray-1);
          padding-top: 1rem;
          font-family: var(--font-primary);
        }
      }
    }
  }
`;
