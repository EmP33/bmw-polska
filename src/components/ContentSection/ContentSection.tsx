import React from "react";

// Icons
import { Car } from "../../assets";
import { ElectricStation } from "../../assets";

// Images
import foto1 from "../../assets/images/foto1_1.png";
import foto2 from "../../assets/images/foto1_2.png";
import foto3 from "../../assets/images/foto2_1.png";
import foto4 from "../../assets/images/foto2_2.png";

// Styles
import { ContentSection } from "./ContentSection.styles";

interface Props {
  type: "blocks" | "text";
}

const ContentSectionComponent: React.FC<Props> = ({ type }) => {
  return (
    <ContentSection type={type}>
      {type === "blocks" ? (
        <div className="content">
          <ElectricStation />
          <h2>FAKTY O NOWYM BMW iX.</h2>
          <div className="content__blocks">
            <div className="block">
              <h6>01</h6>
              <p>Ponad 600 km zasięgu *</p>
            </div>
            <div className="block">
              <h6>02</h6>
              <p>
                Elektryczny napęd na wszystkie koła z dwiema osiami napędzanymi
                <br />
                elektrycznie i ponad 500 KM
                <br /> (370 kW) *
              </p>
            </div>
            <div className="block">
              <h6>03</h6>
              <p>
                Od 0 do 100 km / h <br /> w mniej niż 5 sekund *
              </p>
            </div>
            <div className="block">
              <h6>04</h6>
              <p>
                Ładowanie 120 km <br /> w 10 minut *
              </p>
            </div>
            <div className="block">
              <h6>05</h6>
              <p>
                Publiczne szybkie ładowanie od 10 do 80% w mniej niż <br /> 40
                minut
              </p>
            </div>
          </div>
        </div>
      ) : (
        <div className="content">
          <Car />
          <h2>INNOWACYJNA AERODYNAMIKA BMW iX.</h2>
          <h4>
            BMW iX w każdym milimetrze swojej dynamicznej stylistyki wdraża
            zasadę „forma podąża za funkcją”.
          </h4>
          <p>
            Widać to przede wszystkim po gładkiej karoserii ze zwężoną nadbudową
            kabiny i zredukowanymi fugami oraz po płaskim podwoziu. W połączeniu
            z zamkniętym przodem oraz dyfuzorem z tyłu i niezwykłymi detalami,
            takimi jak wpuszczane klamki czy zintegrowane z listwami szybowymi
            lusterka zewnętrzne, BMW iX osiąga doskonały współczynnik Cx
            wynoszący zaledwie 0,25. Dzięki temu łączy w sobie zasięg i
            wydajność z innowacyjną konstrukcją SAV-a.
          </p>
        </div>
      )}
      {type === "blocks" ? (
        <div className="images">
          <img src={foto1} alt="BMW iX from front side" />
          <img src={foto2} alt="Two cards sideways" />
        </div>
      ) : (
        <div className="images">
          <img src={foto4} alt="BMW iX from back side" />
          <img src={foto3} alt="BMW iX sideways" />
        </div>
      )}
    </ContentSection>
  );
};

export default ContentSectionComponent;
