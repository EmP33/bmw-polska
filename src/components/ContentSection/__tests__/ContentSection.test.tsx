import ContentSection from "../ContentSection";
import ReactShallowRenderer from "react-test-renderer/shallow";

describe("<ContentSection />", () => {
  it("should render ContentSection text type properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ContentSection type="text" dir="ltr" />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("should render ContentSection blocks type properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ContentSection type="blocks" dir="rtl" />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
