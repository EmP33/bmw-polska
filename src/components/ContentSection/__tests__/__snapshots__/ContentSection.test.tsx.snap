// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentSection /> should render ContentSection blocks type properly 1`] = `
<styled.section
  type="blocks"
>
  <div
    className="content"
  >
    <ElectricStation />
    <h2>
      FAKTY O NOWYM BMW iX.
    </h2>
    <div
      className="content__blocks"
    >
      <div
        className="block"
      >
        <h6>
          01
        </h6>
        <p>
          Ponad 600 km zasięgu *
        </p>
      </div>
      <div
        className="block"
      >
        <h6>
          02
        </h6>
        <p>
          Elektryczny napęd na wszystkie koła z dwiema osiami napędzanymi
          <br />
          elektrycznie i ponad 500 KM
          <br />
           (370 kW) *
        </p>
      </div>
      <div
        className="block"
      >
        <h6>
          03
        </h6>
        <p>
          Od 0 do 100 km / h 
          <br />
           w mniej niż 5 sekund *
        </p>
      </div>
      <div
        className="block"
      >
        <h6>
          04
        </h6>
        <p>
          Ładowanie 120 km 
          <br />
           w 10 minut *
        </p>
      </div>
      <div
        className="block"
      >
        <h6>
          05
        </h6>
        <p>
          Publiczne szybkie ładowanie od 10 do 80% w mniej niż 
          <br />
           40 minut
        </p>
      </div>
    </div>
  </div>
  <div
    className="images"
  >
    <img
      alt="BMW iX from front side"
      src="foto1_1.png"
    />
    <img
      alt="Two cards sideways"
      src="foto1_2.png"
    />
  </div>
</styled.section>
`;

exports[`<ContentSection /> should render ContentSection text type properly 1`] = `
<styled.section
  type="text"
>
  <div
    className="content"
  >
    <Car />
    <h2>
      INNOWACYJNA AERODYNAMIKA BMW iX.
    </h2>
    <h4>
      BMW iX w każdym milimetrze swojej dynamicznej stylistyki wdraża zasadę „forma podąża za funkcją”.
    </h4>
    <p>
      Widać to przede wszystkim po gładkiej karoserii ze zwężoną nadbudową kabiny i zredukowanymi fugami oraz po płaskim podwoziu. W połączeniu z zamkniętym przodem oraz dyfuzorem z tyłu i niezwykłymi detalami, takimi jak wpuszczane klamki czy zintegrowane z listwami szybowymi lusterka zewnętrzne, BMW iX osiąga doskonały współczynnik Cx wynoszący zaledwie 0,25. Dzięki temu łączy w sobie zasięg i wydajność z innowacyjną konstrukcją SAV-a.
    </p>
  </div>
  <div
    className="images"
  >
    <img
      alt="BMW iX from back side"
      src="foto2_2.png"
    />
    <img
      alt="BMW iX sideways"
      src="foto2_1.png"
    />
  </div>
</styled.section>
`;
